pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -134,6 +141,20 @@ func (rc *reconciler) syncStates() {
 	rc.attacherDetacher.VerifyVolumesAreAttached(volumesPerNode, rc.actualStateOfWorld)
 }
 
+// hasOutOfServiceTaint returns true if the node has out of service taint present
+// with `NoExecute` effect.
+func (rc *reconciler) hasOutOfServiceTaint(nodeName types.NodeName) (bool, error) {
+	node, err := rc.nodeLister.Get(string(nodeName))
+	if err != nil {
+		return false, err
+	}
+	taintToMatch := &v1.Taint{
+		Key:    v1.TaintNodeOutOfService,
+		Effect: v1.TaintEffectNoExecute,
```

Comment:
again, I don't think you care the effect

Reply:
Done. Fixed it.